odl/tomo/backends/astra_setup.py

```diff
@@ -0,0 +1,581 @@
+        #   'WindowMinX': z_max
+        #   'WindowMaxX': z_max
+        #   'WindowMinY': y_min
+        #   'WindowMaxY': y_min
+        #   'WindowMinZ': x_min
+        #   'WindowMaxZ': x_min
+        vol_geom = astra.create_vol_geom(vol_shp[1], vol_shp[2], vol_shp[0],
+                                         vol_min[2], vol_max[2],
+                                         vol_min[1], vol_max[1],
+                                         vol_min[0], vol_max[0])
+    else:
+        raise ValueError('{}-dimensional volume geometries not supported '
+                         'by ASTRA.'.format(discr_reco.ndim))
+    return vol_geom
+
+
+def astra_geom_to_vec(geometry):
+    """Create vectors for ASTRA projection geometries from `odl.tomo.geometry`.
+
+     The 3D vectors are used to create an ASTRA projection geometry for
+     cone beam geometries ('conve_vec') with helical acquisition curves.
+
+     Additionally, supports creation of 2D and 3D vectors for 'flat_vec' or
+     'parallel3d_vec', respectively.
+
+    Output vectors:
+
+    2d geometry 'fanflat_vec':
+    Each row of vectors corresponds to a single projection, and consists of:
+        ( srcX, srcY, dX, dY, uX, uY )
+        src : the ray source
+        d : the center of the detector
+        u : the vector between the centers of detector pixels 0 and 1
+
+    3d geometry 'parallel3d_vec':
+    Each row of vectors corresponds to a single projection, and consists of:
+        ( rayX, rayY, rayZ, dX, dY, dZ, uX, uY, uZ, vX, vY, vZ )
+        ray : the ray direction
+        d   : the center of the detector
+        u   : the vector from detector pixel (0,0) to (0,1)
+        v   : the vector from detector pixel (0,0) to (1,0)
+
+    3d geometry 'cone_vec':
+    Each row of vectors corresponds to a single projection, and consists of:
+        ( srcX, srcY, srcZ, dX, dY, dZ, uX, uY, uZ, vX, vY, vZ ):
+        src : the ray source
+        d   : the center of the detector
+        u   : the vector from detector pixel (0,0) to (0,1)
+        v   : the vector from detector pixel (0,0) to (1,0)
+
+    Parameters
+    ----------
+    geometry : `Geometry`
+        The `odl.tomo.geometry` instance used to create the ASTRA geometry
+
+    Returns
+    -------
+    vectors : `numpy.ndarray`
+        Numpy array of shape ``(number of angles, 12)``
+    """
+
+    angles = geometry.angle_grid
+    num_angles = geometry.angle_grid.ntotal
+
+    if isinstance(geometry, (CircularConeFlatGeometry,
+                             HelicalConeFlatGeometry)):
+        vectors = np.zeros((num_angles, 12))
+        det_pix_width = geometry.det_grid.stride[0]
+        det_pix_height = geometry.det_grid.stride[1]
+
+        for and_ind in range(num_angles):
+            angle = angles[and_ind][0]
+
+            # source position
+            # TODO: check geometry class for consistency since 'src_position'
+            # and 'det_refpoint' were adopted to use ASTRA cone_vec convention
```

Comment:
What does this comment mean? Adopting to good conventions is all nice, but we need to be consistent internally. These methods should all do "comparable" things across classes, and it's better to translate between different conventions than to adopt conventions from case to case independently.